stdlib/2and3/distutils/version.pyi

```diff
@@ -0,0 +1,35 @@
+# Stubs for distutils.version (Python 2 and 3.5)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+import sys
+from typing import Any
+
+class Version:
+    def __init__(self, vstring=None): ...
+
+    if sys.version_info >= (3,):
```

Comment:
Oh. Well the class in python3 actually has those methods but doesn't in
Python 2.  In python2 they are implemented as __cmp__ in the subclasses.

On Aug 5, 2016 2:31 PM, "Matthias Kramm" <notifications@github.com> wrote:

In stdlib/2and3/distutils/version.pyi
<https://github.com/python/typeshed/pull/450#discussion_r73762483>:

> @@ -0,0 +1,35 @@
> +# Stubs for distutils.version (Python 2 and 3.5)
> +#
> +# NOTE: This dynamically typed stub was automatically generated by stubgen.
> +
> +import sys
> +from typing import Any
> +
> +class Version:
> +    def __init__(self, vstring=None): ...
> +
> +    if sys.version_info >= (3,):

It looks odd that you have to declare these elementary operations (__eq__
etc.), but only for Python 3.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/python/typeshed/pull/450/files/26a3950c3e3718d1ff83854ce34bbd0b80212d7e#r73762483>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABCCAps7FcoRoI9KOEZV8mrkt5Pp9sLyks5qc6uVgaJpZM4JeEiI>
.